libsql/src/sync.rs

```diff
@@ -75,4 +99,46 @@ impl SyncContext {
     pub(crate) fn durable_frame_num(&self) -> u32 {
         self.durable_frame_num
     }
+
+    async fn write_metadata(&mut self) -> Result<()> {
+        let path = format!("{}-info", self.db_path);
+
+        let contents = serde_json::to_vec(&MetadataJson {
+            version: METADATA_VERSION,
+            durable_frame_num: self.durable_frame_num,
+        })
+        .unwrap();
+
+        tokio::fs::write(path, contents).await.unwrap();
```

Comment:
Wouldn't it be safer to write to tmp file and then atomically rename to avoid situation when process crashes in the middle of writting?